File: new-algorithm/src/javaalgo/implementation/boj1476.java
package javaalgo.implementation;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class boj1476 {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());

        int E = Integer.parseInt(st.nextToken());
        int S = Integer.parseInt(st.nextToken());
        int M = Integer.parseInt(st.nextToken());

        int e = 1, s = 1, m = 1;
        int year = 1;

        while (true) {
            if (E == e && S == s && M == m) break;

            e++;
            s++;
            m++;

            if (e == 16) e = 1;
            if (s == 29) s = 1;
            if (m == 20) m = 1;
            year++;
        }

        System.out.println(year);

    }
}

File: new-algorithm/src/kotlinalgo/implementation/boj1476.kt
package kotlinalgo.implementation

import java.io.BufferedReader
import java.io.InputStreamReader
import java.util.StringTokenizer

fun main() = with(BufferedReader(InputStreamReader(System.`in`))) {
    val st = StringTokenizer(readLine())

    var E = st.nextToken().toInt()
    var S = st.nextToken().toInt()
    var M = st.nextToken().toInt()

    var e = 1
    var s = 1
    var m = 1
    var year = 1

    while (true) {
        if (e == E && s == S && m == M) break

        e++
        s++
        m++

        if (e == 16) e = 1
        if (s == 29) s = 1
        if (m == 20) m = 1
        year++
    }

    println(year)
}